scripts/do-release.sh

```diff
@@ -88,7 +88,14 @@ function create_pr() {
     _git commit -a -s -m "${msg}"
     push_to_repo "${branch}"
     to_review=$(dryrun gh pr create --repo "${ORG}/${project}" --head "${branch}" --base "${base_branch}" --title "${msg}" \
-                --label "ready-to-test" --label "e2e-all-k8s" --body "${msg}")
+                --label "ready-to-test" --label "e2e-all-k8s" --body "${msg}" 2>&1)
+
+    # shellcheck disable=SC2181 # The command is too long already, this is more readable
+    if [[ $? -ne 0 ]]; then
```

Comment:
I guess the other reasons the linter complains are related to avoiding potential bugs as the code changes in the future, not only readability:

* Since the command and its status test are decoupled, inserting an innocent command like echo "make finished" after make will cause the if statement to silently start comparing echo's status instead.
* Scripts that run or are called with set -e aka errexit will exit immediately if the command fails, even though they're followed by a clause that handles failure.

https://github-wiki-see.page/m/koalaman/shellcheck/wiki/SC2181

Reply:
> I guess the other reasons the linter complains are related to avoiding potential bugs as the code changes in the future, not only readability:

That's also a valid concern, but readability is not insignificant.

> 
>     * Since the command and its status test are decoupled, inserting an innocent command like echo "make finished" after make will cause the if statement to silently start comparing echo's status instead.

True but we can be careful and bash programmers usually are aware of these pitfalls

> 
>     * Scripts that run or are called with set -e aka errexit will exit immediately if the command fails, even though they're followed by a clause that handles failure.

This doesn't happen in this case so it's not a problem for us, but should it happen we can fix it

> 
> 
> https://github-wiki-see.page/m/koalaman/shellcheck/wiki/SC2181

@skitt WDYT?

Reply:
I think SC2181 is moot, and not just because the command is too long. `$?` reflects the exit code of `dryrun gh pr create ...`, but the latter is run within a command substitution. This means that we can’t use an `if` construct directly with `if dryrun gh pr create ...`.

Reply:
So if there’s anything to fix, it’s the `disable=SC2181` comment ;-).

Reply:
Huh, scratch that, `if to_review=$(...)` would work, so the command length is the reason. (And fine by me.)

Reply:
Okay, thanks @skitt and @mkolesnik.